cl/citations/tests.py

```diff
@@ -355,39 +429,350 @@ def test_disambiguate_citations(self):
 
     def test_make_html(self):
         """Can we make basic HTML conversions properly?"""
-        good_html = (
-            '<pre class="inline">asdf </pre><span class="citation '
-            'no-link"><span class="volume">22</span> <span '
-            'class="reporter">U.S.</span> <span class="page">33</span>'
-            '</span><pre class="inline"> asdf</pre>'
-        )
+        # fmt: off
 
-        # Simple example
-        s = "asdf 22 U.S. 33 asdf"
-        opinion = Opinion(plain_text=s)
-        citations = get_citations(s)
-        new_html = create_cited_html(opinion, citations)
-        self.assertEqual(
-            good_html, new_html,
-        )
+        full_citation_html = ('<pre class="inline">asdf </pre><span class="'
+                              'citation no-link"><span class="volume">22'
+                              '</span> <span class="reporter">U.S.</span> '
+                              '<span class="page">33</span> </span><pre class='
+                              '"inline">asdf</pre>')
+        test_pairs = [
+            # Simple example for full citations
+            ('asdf 22 U.S. 33 asdf', full_citation_html),
+
+            # Using a variant format for U.S. (Issue #409)
+            ('asdf 22 U. S. 33 asdf', full_citation_html),
+
+            # Full citation across line break
+            ('asdf John v. Doe, 123\nU.S. 456, upholding foo bar',
+             '<pre class="inline">asdf John v. Doe, </pre><span class="'
+             'citation no-link"><span class="volume">123</span>\n<span class='
+             '"reporter">U.S.</span> <span class="page">456</span></span><pre'
+             ' class="inline">, upholding foo bar</pre>'),
+
+            # First kind of short form citation (meaningless antecedent)
+            ('asdf. 515 U.S., at 240. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf. </span><span class="volume">515</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">240'
+             '</span></span><pre class="inline">. foobar</pre>'),
+
+            # Second kind of short form citation (meaningful antecedent)
+            ('asdf, 1 U. S., at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf, </span><span class="volume">1</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">2'
+             '</span></span><pre class="inline">. foobar</pre>'),
+
+            # Short form citation with no comma after reporter in original
+            ('asdf, 1 U. S. at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf, </span><span class="volume">1</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">2'
+             '</span></span><pre class="inline">. foobar</pre>'),
```

Comment:
Does this example show a comma being added by our code after `U.S.`? That's probably an issue. Lawyers make big deals about commas.

Reply:
Yes, good catch, I will fix this so it only inserts a comma if there was one in the original text

Reply:
Fixed in b1e76cd